src/core/ext/transport/chttp2/transport/flow_control.cc

```diff
@@ -170,7 +171,7 @@ int64_t TransportFlowControl::target_window() const {
   return static_cast<uint32_t>(
       std::min(static_cast<int64_t>((1u << 31) - 1),
                announced_stream_total_over_incoming_window_ +
-                   target_initial_window_size_));
+                   std::max<int64_t>(1, target_initial_window_size_)));
```

Comment:
why? If this needs to non-zero, let's just change the logic of `target_initial_window_size_` to have a floor of 1

Reply:
transport flow control needs a floor of one, but initial window size can certainly go down to zero
tying them together might be a mistake, but it's a mistake for later